### ISSUE TITLE
Advanced Search - Develop flexible AND/OR capability when chaining simple and group searches

### ISSUE THREAD

**Issue by azerbini** [azerbini/eamena_v3#21]
The user needs to be able to select whether two sets of chained searches are linked by an AND or an OR. Currently, only one global AND/OR may be selected. With this enhancement, each set of two search boxes is linked by an AND/OR operator which the user can modify dynamically via the search UI.

**Comment by azerbini** [comment]
@markdoub & @ahaith do I need to reindex following your latest changes? I presume not?

**Comment by ahaith** [comment]
Thankfully not!

**Comment by azerbini** [comment]
@ahaith The search query seems to be broken, I get a Value Error  'No JSON object could be decoded' with this query:
`"GET /search/resources?page=1&termFilter=%5B%5B%7B%22inverted%22%3Afalse%2C%22type%22%3A%22concept%22%2C%22context%22%3A%2275997a72-c5f5-48a8-a88f-3ba7a28e083d%22%2C%22context_label%22%3A%22Modern%20Country%20Territory%22%2C%22id%22%3A%22Lebanon75997a72-c5f5-48a8-a88f-3ba7a28e083d%22%2C%22text%22%3A%22Lebanon%22%2C%22value%22%3A%227935b516-63f6-47d5-9324-18c6310e9725%22%7D%5D%5D&temporalFilter=%5B%7B%22year_min_max%22%3A%5B%5D%2C%22filters%22%3A%5B%5D%2C%22inverted%22%3Afalse%7D%5D&spatialFilter=%7B%22geometry%22%3A%7B%22type%22%3A%22%22%2C%22coordinates%22%3A%5B%5D%7D%2C%22buffer%22%3A%7B%22width%22%3A%220%22%2C%22unit%22%3A%22m%22%7D%2C%22inverted%22%3Afalse%7D&mapExpanded=false&timeExpanded=false&booleanSearch=and&termFilterAndOr=%5B%22and%22%5D&termFilterGroup=%5B%22No%20group%22%5D&advancedSearch=true&include_ids=true HTTP/1.1"`

**Comment by ahaith** [comment]
There should be a query parameter there with something like &termFilterCombineWithPrev=[True,False].
Do the And/Or toggles show up in the UI?

**Comment by azerbini** [comment]
@ahaith They do, but there are issues with the layout, see below:

![screen shot 2017-10-13 at 13 42 22](https://user-images.githubusercontent.com/19243127/31546723-64255fc0-b01c-11e7-90f3-d0105689b650.png)

**Comment by ahaith** [comment]
Hmm, that doesn't look right. I think we'll need Markos' help to fix this on Monday.

**Comment by markdoub** [comment]
Hi Andrea,

Unfortunately I am not in front of my computer. The only thing I can think now is that maybe you are not logged (but you probably are) or the query is generated by the is code before the latest commits.
Can you click search to start a new blank search and try again?

Regards,
Markos


-------- Original Message --------
Subject: Re: [azerbini/eamena_v3] Advanced Search - Develop flexible AND/OR capability when chaining simple and group searches (#21)
From: Andy Haith
To: azerbini/eamena_v3
CC: Markos Ntoumpanakis ,Mention


Hmm, that doesn't look right. I think we'll need Markos' help to fix this on Monday.

**Comment by azerbini** [comment]
@markdoub Thanks for this. I did try to reset the query, but the same error is still there. There seems to be some issue with the UI too (as the above screenshot shows). Cheers

**Comment by markdoub** [comment]
Good morning Andrea,

I think that we pushed the “term filter combination” changes to your advanced-search branch instead of the advanced-group-search-fix-master we usually use, maybe this is way it was not working. I just pushed the changes for the “term filter combination” search to the advanced-group-search-fix-master. Is it possible to test it from there?

Regards,
Markos


On 13 Oct 2017, at 14:39, Andrea Zerbini <notifications@github.com<mailto:notifications@github.com>> wrote:


@markdoub<https://github.com/markdoub> Thanks for this. I did try to reset the query, but the same error is still there. There seems to be some issue with the UI too (as the above screenshot shows). Cheers
